File: doordeck-sdk/src/mingwMain/resources/csharp/Model/ResultData.cs
namespace Doordeck.Headless.Sdk.Model
{
    public class ResultData<T>
    {
        public SuccessResultData<T>? Success { get; set; } = null;
        public FailedResultData? Failure { get; set; } = null;

        public bool IsSuccess => Success is not null;
        public bool IsFailure => Failure is not null;
    }

    public class SuccessResultData<T>
    {
        public T? Result { get; set; } = default;
    }

    public class FailedResultData
    {
        public string ExceptionType { get; set; } = string.Empty;
        public string ExceptionMessage { get; set; } = string.Empty;
    }
}

File: doordeck-sdk/src/mingwMain/kotlin/com/doordeck/multiplatform/sdk/model/ResultData.kt
package com.doordeck.multiplatform.sdk.model

import kotlinx.serialization.Serializable

@Serializable
data class ResultData<T>(
    val success: SuccessResultData<T>? = null,
    val failure: FailedResultData? = null
)

@Serializable
data class SuccessResultData<T>(
    val result: T? = null
)

@Serializable
data class FailedResultData(
    val exceptionType: String,
    val exceptionMessage: String
)